chapters/arrays.tex

```diff
@@ -55,6 +55,7 @@ \section{Array Declarations}\label{array-declarations}
 \item
   Binding equations of the form \lstinline!x1 = x2! are allowed for arrays independent of whether the index types of dimensions are subtypes of \lstinline!Integer!, \lstinline!Boolean!, or enumeration types.
 \end{itemize}
+Operations constructing arrays (e.g., matrix multiplication, indexing, array constructors) only generate arrays indexed by \lstinline!Integer!.
```

Comment:
If not placed in a non-normative section, it can at least be formulated as a consequence with explanations and references to relevant parts of the document, rather than an added restriction.  This would also give us a better chance at seeing what lies behind this statement.

Reply:
Thinking some more about #3410, I suggest that we skip this sentence for now, and instead consider including it in a PR for #3410:
```suggestion
```